metagpt/actions/invoice_ocr.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+# _*_ coding: utf-8 _*_
+
+"""
+@Time    : 2023/9/21 18:10:20
+@Author  : Stitch-z
+@File    : invoice_ocr.py
+@Describe : Actions of the invoice ocr assistant.
+"""
+
+import os
+import zipfile
+from enum import Enum
+from pathlib import Path
+
+import pandas as pd
+
+from typing import Dict, List
+from paddleocr import PaddleOCR
+
+from metagpt.actions import Action
+from metagpt.const import INVOICE_OCR_TABLE_PATH
+from metagpt.logs import logger
+from metagpt.prompts.invoice_ocr import EXTRACT_OCR_MAIN_INFO_PROMPT, REPLY_OCR_QUESTION_PROMPT
+from metagpt.utils.common import OutputParser
+from metagpt.utils.file import File
+
+
+class FileExtensionType(Enum):
+    """Enum representing file extensions and their associated types.
+    Each enum member consists of a tuple containing the file extension and its associated type.
+
+    """
+
+    ZIP = (".zip", "zip")
+    PDF = (".pdf", "pdf")
+    PNG = (".png", "png")
+    JPG = (".jpg", "jpg")
+
+    @classmethod
+    def get_extension_list(cls) -> List[str]:
+        """Get a list of file extensions.
+
+        Returns:
+            A list of file extensions as strings.
+        """
+        return [ext.value[0] for ext in cls]
+
+    @classmethod
+    def get_type_list(cls) -> List[str]:
+        """Get a list of file types.
+
+        Returns:
+            A list of file types as strings.
+        """
+        return [ext.value[1] for ext in cls]
+
+
+class InvoiceOCR(Action):
+    """Action class for performing OCR on invoice files, including zip, PDF, png, and jpg files.
+
+    Args:
+        name: The name of the action. Defaults to an empty string.
+        language: The language for OCR output. Defaults to "ch" (Chinese).
+
+    """
+
+    def __init__(self, name: str = "", *args, **kwargs):
+        super().__init__(name, *args, **kwargs)
+
+    @staticmethod
+    async def _check_file_type(filename: str) -> str:
+        """Check the file type of the given filename.
+
+        Args:
+            filename: The name of the file.
+
+        Returns:
+            The file type based on FileExtensionType enum.
+
+        Raises:
+            Exception: If the file format is not zip, pdf, png, or jpg.
+        """
+        file_ext = None
+        for ext in FileExtensionType:
```

Comment:
may an  support_ext = [".zip", ".jpg",xxx]  is more simple. and you can use Path(xx).suffix to have a judge. The enum make it complex.